File: data/js/AI/A10.js
// AI-generated encryption example
// Demonstrates AES encryption and SHA256 hashing

const crypto = require('crypto');

// AES encryption
function encrypt(text, key) {
    const cipher = crypto.createCipheriv('aes-128-cbc', Buffer.from(key), Buffer.alloc(16, 0));
    let encrypted = cipher.update(text, 'utf8', 'hex');
    encrypted += cipher.final('hex');
    return encrypted;
}

// AES decryption
function decrypt(enc, key) {
    const decipher = crypto.createDecipheriv('aes-128-cbc', Buffer.from(key), Buffer.alloc(16, 0));
    let dec = decipher.update(enc, 'hex', 'utf8');
    dec += decipher.final('utf8');
    return dec;
}

// SHA256 hashing
function hash(text) {
    return crypto.createHash('sha256').update(text).digest('hex');
}

// Example usage
const key = crypto.randomBytes(16);
const message = "This is secret";

const encrypted = encrypt(message, key);
console.log("Encrypted:", encrypted);

const decrypted = decrypt(encrypted, key);
console.log("Decrypted:", decrypted);

console.log("SHA256 hash:", hash(message));

File: data/java/AI/A10.java
import javax.crypto.Cipher;
import javax.crypto.KeyGenerator;
import javax.crypto.SecretKey;
import javax.crypto.spec.SecretKeySpec;
import java.util.Base64;

/**
 * AI-generated encryption/decryption utility
 * Demonstrates AES symmetric encryption in Java
 */
public class EncryptionAI {

    // Generate a new AES secret key
    public static SecretKey generateKey() throws Exception {
        KeyGenerator keyGen = KeyGenerator.getInstance("AES");
        keyGen.init(128); // AES-128
        return keyGen.generateKey();
    }

    // Encrypt plain text using AES key
    public static String encrypt(String plainText, SecretKey key) throws Exception {
        Cipher cipher = Cipher.getInstance("AES");
        cipher.init(Cipher.ENCRYPT_MODE, key);
        byte[] encrypted = cipher.doFinal(plainText.getBytes());
        return Base64.getEncoder().encodeToString(encrypted);
    }

    // Decrypt cipher text using AES key
    public static String decrypt(String cipherText, SecretKey key) throws Exception {
        Cipher cipher = Cipher.getInstance("AES");
        cipher.init(Cipher.DECRYPT_MODE, key);
        byte[] decoded = Base64.getDecoder().decode(cipherText);
        byte[] decrypted = cipher.doFinal(decoded);
        return new String(decrypted);
    }

    public static void main(String[] args) {
        try {
            SecretKey key = generateKey();
            String text = "This is a secret message.";

            String encrypted = encrypt(text, key);
            System.out.println("Encrypted: " + encrypted);

            String decrypted = decrypt(encrypted, key);
            System.out.println("Decrypted: " + decrypted);

        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
